actor/src/lib.rs

```diff
@@ -160,12 +161,15 @@ impl<A: Actor> Addr<A> {
     {
         let envelope = SyncEnvelopeProxy::pack(message, None);
         let sender = self.sender.clone();
-        tokio::task::spawn(async move {
-            // TODO: propagate the error.
-            if let Err(error) = sender.send(envelope).await {
-                iroha_logger::error!(%error, "Error sending actor message");
+        // TODO: BUG: remove deadlock from iroha (probably issue inside of `iroha_p2p` crate) and remove this task::spawn
```

Comment:
Can you report this bug as an issue?

Reply:
There is also a FIXME on top of this function. I think it can be removed, as you explained there is no waiting here.

Reply:
[Done](https://github.com/hyperledger/iroha/issues/1649)